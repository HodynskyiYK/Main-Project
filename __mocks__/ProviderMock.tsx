import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}