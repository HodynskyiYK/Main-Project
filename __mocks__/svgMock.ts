// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock