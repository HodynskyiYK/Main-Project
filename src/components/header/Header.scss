

.header {
  background: $black_color url("../../assets/header-bg.png") no-repeat center/cover;
  padding-top: 1.25rem;
  padding-bottom: 1.25rem;
  margin-bottom: 0.625rem;

  .logo {
    color: $red_color;
    font: 300 1.25rem/1.2em $main_font;

    b {
      font-weight: 900
    }
  }
}