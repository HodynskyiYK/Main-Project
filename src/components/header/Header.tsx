import React, {FC} from 'react'
import {AddMovieButton} from '../add-movie'
import {FindYourMovie} from '../find-your-movie'

export const Header: FC = () => {

    return (
        <header className="header container">
            <div className="row align-items-start justify-content-between">
                <div className="col">
                    <span className="logo">
                        <b>netflix</b>
                        roulette
                    </span>
                </div>
                <div className="col">
                    <AddMovieButton/>
                </div>
                <div className="col col-12">
                    <FindYourMovie/>
                </div>
            </div>
        </header>
    )
}
