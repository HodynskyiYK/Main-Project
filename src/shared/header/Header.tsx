import React, {FC} from 'react'
import classnames from 'classnames'
import {AddMovie} from '../../pages/Home/components/add-movie'
import {FindYourMovie} from '../../pages/Home/components/find-your-movie'
import { Logo } from '../../components/logo'
import { useMovieDetailsContext } from '../../context/movie-details-context'
import styles from './Header.module.scss'

export const Header: FC = () => {
    const {movieDetailsState} = useMovieDetailsContext()

    if (movieDetailsState) {
        return null
    }

    return (
        <header className={classnames('container', styles.header)}>
            <div className={classnames('row', 'align-items-start', 'justify-content-between')}>
                <div className={classnames('col')}>
                    <Logo/>
                </div>
                <div className={classnames('col')}>
                    <AddMovie/>
                </div>
                <div className={classnames('col', 'col-12')}>
                    <FindYourMovie/>
                </div>
            </div>
        </header>
    )
}
