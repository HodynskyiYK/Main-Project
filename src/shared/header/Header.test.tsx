import React from 'react'
import { render } from '@testing-library/react'
import { Header } from './Header'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('Header tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock><Header/></ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})