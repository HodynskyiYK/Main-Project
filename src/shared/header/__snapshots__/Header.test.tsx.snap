// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests To match snapshot 1`] = `
<DocumentFragment>
  <header
    class="container headerContainer headerContent"
  >
    <div
      class="row align-items-start justify-content-between"
    >
      <div
        class="col"
      >
        <span
          class="Logo"
        >
          <b>
            netflix
          </b>
          roulette
        </span>
      </div>
      <div
        class="col"
      >
        <button
          class="btn addBtn"
        >
          + add movie
        </button>
      </div>
      <div
        class="col col-12"
      >
        <div
          class="findYourMovie"
        >
          <p
            class="h1 h1"
          >
            Find your movie
          </p>
          <form
            class="searchForm"
          >
            <div
              class="formGroup"
            >
              <div
                class="findInput"
              >
                <div
                  class="inputField"
                >
                  <input
                    class="field"
                    id="find-movie"
                    name="find-movie"
                    placeholder="What do you want to watch?"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <button
                class="btn searchBtn"
              >
                Search
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
